src/suffer.cpp

```diff
@@ -801,71 +834,95 @@ void Character::suffer_from_albinism()
-            if( !affected_bp.test( bp ) || !covered.test( bp ) ) {
-                continue;
-            }
-            //percent of "not covered skin"
-            float p = 1.0 - i.get_coverage() / 100.0;
-            open_percent[bp] = open_percent[bp] * p;
-        }
-    }
 
-    const float COVERAGE_LIMIT = 0.01;
-    body_part max_affected_bp = num_bp;
-    float max_affected_bp_percent = 0;
-    int count_affected_bp = 0;
-    for( const std::pair<const body_part, float> &it : open_percent ) {
-        const body_part &bp = it.first;
-        const float &p = it.second;
+    std::map<bodypart_id, float> bp_exposure = bodypart_exposure();
 
-        if( p <= COVERAGE_LIMIT ) {
+    // Minimum exposure threshold for pain
+    const float MIN_EXPOSURE = 0.01f;
+    // Count how many body parts are above the threshold
+    int count_affected_bp = 0;
+    // Get the most exposed body part, and how exposed it is.  This is to tell the player what body
+    // part is most irritated by sun, so they know what needs to be covered up better.
+    bodypart_id most_exposed_bp;
+    float max_exposure = 0.0f;
+    // Check each bodypart with exposure above the minimum
+    for( const std::pair<const bodypart_id, float> &bp_exp : bp_exposure ) {
+        const float exposure = bp_exp.second;
+        // Skip minimally-exposed parts, and skip the eyes (handled by sunglasses)
+        if( exposure <= MIN_EXPOSURE || bp_exp.first == bodypart_id( "eyes" ) ) {
             continue;
         }
         ++count_affected_bp;
-        if( max_affected_bp_percent < p ) {
-            max_affected_bp_percent = p;
-            max_affected_bp = bp;
-        }
-    }
-    if( count_affected_bp > 0 && max_affected_bp != num_bp ) {
-        //Check if both arms/legs are affected
-        int parts_count = 1;
-        body_part other_bp = static_cast<body_part>( bp_aiOther[max_affected_bp] );
-        body_part other_bp_rev = static_cast<body_part>( bp_aiOther[other_bp] );
-        if( other_bp != other_bp_rev ) {
-            const auto found = open_percent.find( other_bp );
-            if( found != open_percent.end() && found->second > COVERAGE_LIMIT ) {
-                ++parts_count;
-            }
+        if( exposure > max_exposure ) {
+            max_exposure = exposure;
+            most_exposed_bp = bp_exp.first;
         }
-        std::string bp_name = body_part_name( max_affected_bp, parts_count );
-        if( count_affected_bp > parts_count ) {
-            bp_name = string_format( _( "%s and other body parts" ), bp_name );
-        }
-        add_msg_if_player( m_bad, _( "The sunlight is really irritating your %s." ), bp_name );
+    }
 
-        //apply effects
-        if( has_effect( effect_sleep ) && !has_effect( effect_narcosis ) ) {
-            wake_up();
+    // If all body parts are protected, there is no suffering
+    if( count_affected_bp == 0 || most_exposed_bp == bodypart_str_id( "num_bp" ) ) {
```

Comment:
Don't compare to `num_bp`, just use `most_exposed_bp` as if it was a `bool`:
```suggestion
    if( count_affected_bp == 0 || !most_exposed_bp ) {
```